### PR TITLE
MadMiner v0.8.0

### PR DESCRIPTION
**Pull request by johannbrehmer**
WIP

### CODE REVIEW

**Comment by cranmer** [comment]
Hey @johannbrehmer can you summarize where this left off?

**Comment by johannbrehmer** [comment]
Hi, this is in principle ready to be merged. The only question is whether we already want to give up Python 2 support (remember that the main MadGraph version still only supports Python 2...). I don't have a strong preference, but I'm leaning to wards with this merge until the main MadGraph releases run on Python 3.

**Comment by matthewfeickert** [comment]
> I don't have a strong preference, but I'm leaning to wards with this merge until the main MadGraph releases run on Python 3.

Coming here from Issue #452, but I'm not sure what you mean by "the main MadGraph". [MadGraph5_aMC-NLO has been supported on Python 3 for a long time now](https://github.com/scailfin/MadGraph5_aMC-NLO) and [Python 2 support is being officially dropped in December 2021](https://launchpad.net/mg5amcnlo/+announcement/22860). 

Do you have some requirements that keep you pinned to an older Python 2 only version of MadGraph5_aMC-NLO? If not, please drop Python 2 support for everything.

**Comment by johannbrehmer** [comment]
Fantastic, I totally missed that May 2020 announcement! Then I'm totally in favour of dropping Python 2 support now.

Unfortunately, I can't really promise to commit any time to take care of this. But if anyone else can update this MR, ping me and I'll review it and take care of the release and pip version.

**Comment by Sinclert** [comment]
Given that I was involved in the creation of the `python3-only` branch, I volunteer to do it. However, I do not have `write` permissions on this repository (nor in any of the [DIANA-HEP](https://github.com/diana-hep)) organization.

Could anyone invite me so I can fix the PR conflicts?

**Comment by matthewfeickert** [comment]
> Fantastic, I totally missed that May 2020 announcement! Then I'm totally in favour of dropping Python 2 support now.

Great to hear. :rocket: 

> Unfortunately, I can't really promise to commit any time to take care of this. But if anyone else can update this MR, ping me and I'll review it and take care of the release and pip version.
> Given that I was involved in the creation of the `python3-only` branch, I volunteer to do it. However, I do not have `write` permissions on this repository (nor in any of the [DIANA-HEP](https://github.com/diana-hep)) organization.

Sure, I understand @johannbrehmer. Beyond my out-of-the-blue drive by PRs and Issues I don't think I really have the time to contribute meaningfully, but @Sinclert is awesome and on this &mdash; thank you Sinclert! :+1: 
 
> Could anyone invite me so I can fix the PR conflicts?

I also don't have any permissions on DIANA-HEP, so I think we'll need to ping [people that publicly have admin rights](https://github.com/orgs/diana-hep/people). @cranmer @jpivarski @pelmer can you give @Sinclert permissions please?

**Comment by johannbrehmer** [comment]
@Sinclert thanks so much for all your work! I invited you to have write access. Please let me know if that worked. Ping me if you need more permissions for anything.

**Comment by Sinclert** [comment]
@johannbrehmer @matthewfeickert @cranmer 👋🏻 

I have rebased the `master` branch into the `python3-only` branch solving all the conflicts, while preserving the changes introduced in the `master` branch since I created my fork (from where this PR derives).

This is all good for review! 👀

### Notes:
- I would recommend reading the description of [my original PR](https://github.com/diana-hep/madminer/pull/438), to get a sense of what to find in each commit.
- I would recommend **not** to continue development on `master` until this PR is merged (otherwise, conflicts will appear).

**Comment by johannbrehmer** [comment]
You're a saint!

Tagging @KlingFelix also in case he wants to have a look.

**Comment by Sinclert** [comment]
I just fixed some Python2 references still existing throughout the code base (`python2_override` argument), and updated the [installation.md](https://github.com/diana-hep/madminer/blob/919811debc4745a3194a78ce6ff8cca3894fceaf/docs/installation.md) guide to update both MadGraph and Python minimum versions (now MG 2.8.0+, and Python 3.6+).

@KlingFelix did you have time to review the changes?

If we are good to go, I would suggest merging PR https://github.com/diana-hep/madminer/pull/455 first (leaving MadMiner `0.7.X` unbroken), and then merging this one.

**Comment by Sinclert** [comment]
Hi @johannbrehmer , @KlingFelix  👋🏻 

Just curious, is any of you looking at this PR? Because if you are not, I would suggest merging it and move from there. These changes has been stuck in review process since **October 2020**...

**Comment by johannbrehmer** [comment]
Hi @Sinclert , thanks for the reminder and sorry for the radio silence! I'm afraid that I can't really help with MadMiner development these days. In the future please feel free to just make executive decisions (together with Felix, Kyle etc).

**Comment by johannbrehmer** [comment]
The only thing that only I have the permission for currently is pushing to pypi, but I'm happy to add more users to it! Just send me an email with your username on there. (But I'm also happy to keep pushing to pypi for every new release, that's zero effort.)
